protocol/x/clob/keeper/process_proposer_matches_events.go

```diff
@@ -109,77 +115,77 @@ func (k Keeper) AddDeliveredConditionalOrderId(ctx sdk.Context, orderId types.Or
 // AddDeliveredCancelledOrderId saves a cancelled order id to the memstore for processing
 // in the next PrepareCheckState. The order of additions is not maintained.
 func (k Keeper) AddDeliveredCancelledOrderId(ctx sdk.Context, orderId types.OrderId) {
-	k.SetUnorderedOrderId(ctx, types.DeliveredCancelKeyPrefix, orderId)
+	k.SetUnorderedOrderId(ctx, ctx.KVStore(k.memKey), types.DeliveredCancelKeyPrefix, orderId)
 }
 
 // GetDeliveredLongTermOrderIds gets the ordered list of delivered long term order ids from the memstore.
 func (k Keeper) GetDeliveredLongTermOrderIds(ctx sdk.Context) []types.OrderId {
-	return k.GetOrderIds(ctx, types.OrderedDeliveredLongTermOrderKeyPrefix)
+	return k.GetOrderIds(ctx, ctx.KVStore(k.memKey), types.OrderedDeliveredLongTermOrderKeyPrefix)
 }
 
 // GetDeliveredConditionalOrderIds gets the ordered list of delivered conditional order ids from the memstore.
 func (k Keeper) GetDeliveredConditionalOrderIds(ctx sdk.Context) []types.OrderId {
-	return k.GetOrderIds(ctx, types.OrderedDeliveredConditionalOrderKeyPrefix)
+	return k.GetOrderIds(ctx, ctx.KVStore(k.memKey), types.OrderedDeliveredConditionalOrderKeyPrefix)
 }
 
 // GetDeliveredCancelledOrderIds gets the unordered list of delivered cancelled order ids from the memstore.
 func (k Keeper) GetDeliveredCancelledOrderIds(ctx sdk.Context) []types.OrderId {
-	return k.GetOrderIds(ctx, types.DeliveredCancelKeyPrefix)
+	return k.GetOrderIds(ctx, ctx.KVStore(k.memKey), types.DeliveredCancelKeyPrefix)
 }
 
 // HasDeliveredCancelledOrderId returns the existence of an order id in the memstore's list of delivered
 // cancelled order ids.
 func (k Keeper) HasDeliveredCancelledOrderId(ctx sdk.Context, orderId types.OrderId) bool {
-	return k.HasUnorderedOrderId(ctx, types.DeliveredCancelKeyPrefix, orderId)
+	return k.HasUnorderedOrderId(ctx, ctx.KVStore(k.memKey), types.DeliveredCancelKeyPrefix, orderId)
 }
 
-func (k Keeper) ResetOrderedOrderIds(ctx sdk.Context, keyPrefix string, indexKey string) {
-	memStore := ctx.KVStore(k.memKey)
-	store := prefix.NewStore(memStore, []byte(keyPrefix))
-	it := store.Iterator(nil, nil)
+func (k Keeper) ResetOrderedOrderIds(
+	ctx sdk.Context, store storetypes.KVStore, keyPrefix string, indexKey string,
+) {
+	prefixStore := prefix.NewStore(store, []byte(keyPrefix))
+	it := prefixStore.Iterator(nil, nil)
 	for ; it.Valid(); it.Next() {
-		store.Delete(it.Key())
+		prefixStore.Delete(it.Key())
 	}
-	memStore.Delete([]byte(indexKey))
+	store.Delete([]byte(indexKey))
 }
 
-func (k Keeper) ResetUnorderedOrderIds(ctx sdk.Context, keyPrefix string) {
-	memStore := ctx.KVStore(k.memKey)
-	store := prefix.NewStore(memStore, []byte(keyPrefix))
-	it := store.Iterator(nil, nil)
+func (k Keeper) ResetUnorderedOrderIds(ctx sdk.Context, store storetypes.KVStore, keyPrefix string) {
```

Comment:
For my understanding what's the motivation behind changing this interface?

Reply:
I'm using it with persisted KVStore in this PR.